### ISSUE TITLE
Activating a switch or radio with Semantics.onTap do not gives feedback about state with TalkBack

### ISSUE THREAD

**Issue by Zigotote** [flutter/flutter#155298]
### Steps to reproduce

1. Turn on TalkBack on an Android device
2. Create a page with a Text widget and a Switch, grouped inside a Semantics widget
3. Focus the Semantic group, containing the text + switch
4. The state of this widget is read by the Talkback  : "unchecked, text, double tap to activate"
5. Double tap

### Expected results

The new state of the widget is read by the Talkback : "checked"

### Actual results

Nothing happens. This is also the case with a Radio widget

However, the state is read : 
- on an iOS device, with VoiceOver
- if the switch is alone and has no custom semantics on top of it

I found 2 issues about this. Even if they are solved it doesn't work in my case : 
- https://github.com/flutter/flutter/issues/17775
- https://github.com/flutter/flutter/issues/14092

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

class MyPage extends StatefulWidget {
  const MyPage({super.key});

  @override
  State<MyPage> createState() => _MyPageState();
}

class _MyPageState extends State<MyPage> {
  bool _isActivated = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Semantics(
          container: true,
          label: "Activate my switch",
          toggled: _isActivated,
          enabled: true,
          onTap: () => _saveState(!_isActivated),
          excludeSemantics: true,
          child: GestureDetector(
            onTap: () => _saveState(!_isActivated),
            excludeFromSemantics: true,
            behavior: HitTestBehavior.opaque,
            child: Row(
              children: [
                const Expanded(child: Text("Activate my switch")),
                Switch(value: _isActivated, onChanged: _saveState),
              ],
            ),
          ),
        ),
      ),
    );
  }

  _saveState(bool consent) {
    setState(() {
      _isActivated = consent;
    });
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![Capture d’écran 2024-09-17 à 16 10 20](https://github.com/user-attachments/assets/17c722b2-6f86-48fc-ac17-abfa791d8da2)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.6.1 23G93 darwin-x64, locale fr-FR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.2.4)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I was able to replicate the reported behavior on Android. Below is observation:

1. Android - talkback : `<text> switch, double tap to toggle`, upon toggling, the state is not announced.
2.  iOS - Voiceover: <text> switch button, off, double tap to toggle, upon togglgine, the state is announced as `on`.

```
Stable : 3.24.3
Master : 3.26.0-1.0.pre.152
```

**Comment by chunhtai** [comment]
My guess is that it expects synchronizous update to the state. A bandage fix is to update the local state synchronously

**Comment by chunhtai** [comment]
radio button seems to be working for me
```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [Radio].

void main() => runApp(const RadioExampleApp());

class RadioExampleApp extends StatelessWidget {
  const RadioExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('Radio Sample')),
        body: const Center(
          child: RadioExample(),
        ),
      ),
    );
  }
}

enum SingingCharacter { lafayette, jefferson }

class RadioExample extends StatefulWidget {
  const RadioExample({super.key});

  @override
  State<RadioExample> createState() => _RadioExampleState();
}

class _RadioExampleState extends State<RadioExample> {
  SingingCharacter? _character = SingingCharacter.lafayette;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        ListTile(
          title: const Text('Lafayette'),
          leading: Radio<SingingCharacter>(
            value: SingingCharacter.lafayette,
            groupValue: _character,
            onChanged: (SingingCharacter? value) {
              setState(() {
                _character = value;
              });
            },
          ),
        ),
        ListTile(
          title: const Text('Thomas Jefferson'),
          leading: Radio<SingingCharacter>(
            value: SingingCharacter.jefferson,
            groupValue: _character,
            onChanged: (SingingCharacter? value) {
              setState(() {
                _character = value;
              });
            },
          ),
        ),
      ],
    );
  }
}
```

**Comment by chunhtai** [comment]
both switch and radio worked fine, if you want to create your own semantics, you should use 
`Feedback.forTap(context)` in savestate for announcement
